### PR TITLE
Update free-courses-hi.md

### PR DESCRIPTION
**Pull request by SoulNikhar**
## What does this PR do?
- KnowledgeGate by Sanchit Sir is one of the best channel for compiler design. 

### Description
- KnowledgeGate by Sanchit Sir is one of the best channel for compiler design. 

### Why is this valuable (or not)?
- For learning 

### How do we know it's really free?
- It's on youtube

### For book lists, is it a book? For course lists, is it a course? etc.
- It's a course

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by Ajay-singh1** [comment]
Follow Alphabetical order sequencing and proper indentation.

**Comment by SoulNikhar** [comment]
> Follow Alphabetical order sequencing and proper indentation.

Done

**Comment by Ajay-singh1** [comment]
LGTM!

**Review by sohansai — COMMENTED** [review]
We prefer 'playlist' links for YouTube contributions, and the link you provided is a single video. 

The link you provided seems different from the title.

Are you referring to this link:

https://www.youtube.com/playlist?list=PLmXKhU9FNesSmu-_DKC7APRoFkaQvGurx

**Comment by SoulNikhar** [comment]
Hii.. @sohansai , Right I am referring to below link 

https://www.youtube.com/playlist?list=PLmXKhU9FNesSmu-_DKC7APRoFkaQvGurx

**Comment by sohansai** [comment]
You've made an alphabetical order error. Please correct it and follow alphabetical ordering. Since this is your first time, feel free to make good contributions in the future. Make sure to check the 'free-programming-books-lint' output for linter errors.

**Comment by SoulNikhar** [comment]
Sorry I am unable to identify what to correct. I know it showing to correct alphabetical ordering but I am unable to do so because it showing error in free-programming-books-bn.md   but I had made changes in free-courses-hi.md

Error is showing in free-programming-books-bn.md

image

But i made a change in free-courses-hi.md

image


Please @sohansai  help me out

**Comment by SoulNikhar** [comment]
@sohansai  Initially i got errors of alphabetical ordering but line number was also mentioned clearly 
eg: Previous errors

image

But now I am unable to find my errors 
so Please help me out

File: courses/free-courses-hi.md [hunk 1]
```diff
@@ -213,6 +213,7 @@
 
 * [Compiler Design](https://www.youtube.com/playlist?list=PL9FuOtXibFjVR-87LcU-DS-9TJcbG97_p) - Abhishek Sharma
 * [Compiler Design](https://www.youtube.com/playlist?list=PL23dd-8zssJBiyntds3X1sVWUDeb0Aa1N) - Start Practicing
+* [Compiler Design](https://www.youtube.com/playlist?list=PLmXKhU9FNesSmu-_DKC7APRoFkaQvGurx) - KnowledgeGate
```
Comment (eshellman) [inline courses/free-courses-hi.md:216]: Don't invent titles! Use the title given by the resource.
Please add creator names/channel name where appropriate.

**Comment by sohansai** [comment]
No all CAPS

**Comment by SoulNikhar** [comment]
> No all CAPS

Removed all CAPS

**Comment by SoulNikhar** [comment]
Hii... @eshellman Is any more changes required??

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!
(Sorry for delay, your question yesterday pushed the PR to the end of my queue.)

**Pull request merged**